src/compression.h

```diff
@@ -0,0 +1,25 @@
+#ifndef TINC_COMPRESSION_H
+#define TINC_COMPRESSION_H
+
+typedef enum compression_level_t {
+	COMPRESS_NONE = 0,
+
+	COMPRESS_ZLIB_1 = 1,
+	COMPRESS_ZLIB_2 = 2,
+	COMPRESS_ZLIB_3 = 3,
+	COMPRESS_ZLIB_4 = 4,
+	COMPRESS_ZLIB_5 = 5,
+	COMPRESS_ZLIB_6 = 6,
+	COMPRESS_ZLIB_7 = 7,
+	COMPRESS_ZLIB_8 = 8,
+	COMPRESS_ZLIB_9 = 9,
+
+	COMPRESS_LZO_LO = 10,
+	COMPRESS_LZO_HI = 11,
+
+	COMPRESS_LZ4 = 12,
+
+	COMPRESS_GUARD = INT_MAX, /* ensure that sizeof(compression_level_t) == sizeof(int) */
+} compression_level_t;
```

Comment:
I'm getting compiler warnings myself now:

```
control.c:133:29: warning: format ‘%d’ expects argument of type ‘int *’, but argument 3 has type ‘compression_level_t *’ [-Wformat=]
  133 |   sscanf(request, "%*d %*d %d", &c->outcompression);
      |                            ~^   ~~~~~~~~~~~~~~~~~~
      |                             |   |
      |                             |   compression_level_t *
      |                             int *
```

And:

```
protocol_key.c:404:21: warning: format ‘%d’ expects argument of type ‘int *’, but argument 9 has type ‘compression_level_t *’ [-Wformat=]
  404 |  if(sscanf(request, "%*d "MAX_STRING" "MAX_STRING" "MAX_STRING" %d %d %zu %d "MAX_STRING" "MAX_STRING,
      |                     ^~~~~~
```

So for `printf()` and `scanf()`, we can't dereference pointers to `enum`s. I like the `enum` and I think we should keep it, but maybe revert the type of the variables storing compression levels back to `int`.

Reply:
Rats. For some reason, neither clang nor gcc were reporting these warnings on my system.

I reverted compression fields to `int` and added `-Werror` to the [static](https://github.com/gsliepen/tinc/runs/3218917811?check_suite_focus=true#step:12:855) analysis [job](https://github.com/hg/tinc/runs/3218249344?check_suite_focus=true) so as not to burden you with such trivia in the future. We probably won't be able to add it everywhere because of [old compilers](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53119) that nag about `struct type_t xxx = { 0 };`, and macOS which deprecates something every release.

Failing on `-Werror` won't affect other checks.